applications/ShallowWaterApplication/custom_strategies/schemes/residualbased_incrementalupdate_wetting_scheme.h

```diff
@@ -0,0 +1,246 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+#ifndef KRATOS_RESIDUALBASED_INCREMENTALUPDATE_WETTING_SCHEME_H_INCLUDED
+#define KRATOS_RESIDUALBASED_INCREMENTALUPDATE_WETTING_SCHEME_H_INCLUDED
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "processes/process.h"
+#include "solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h"
+
+
+namespace Kratos
+{
+///@addtogroup ShallowWaterApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+*/
+template<class TSparseSpace, class TDenseSpace >
+class ResidualBasedIncrementalUpdateWettingScheme : public ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace, TDenseSpace>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of ResidualBasedIncrementalUpdateWettingScheme
+    KRATOS_CLASS_POINTER_DEFINITION(ResidualBasedIncrementalUpdateWettingScheme);
+
+    typedef ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace, TDenseSpace> BaseType;
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// @brief Default constructor.
+    ResidualBasedIncrementalUpdateWettingScheme()
+        : BaseType()
+    {}
+
+    /// @brief Constructor with wetting drying model.
+    ResidualBasedIncrementalUpdateWettingScheme(Process::Pointer pWettingModel)
```

Comment:
The name is a bit confusing 
Model <=> Process

Reply:
I'm using Process as a base class...
This is experimental. If in the future I add more wetting models, I will add a specific base class for the models. Do you agree?